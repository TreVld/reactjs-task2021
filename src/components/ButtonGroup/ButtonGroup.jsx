import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import './ButtonGroup.scss'

function ButtonGroup (props) {
  const className = classNames('btn-group', {
    'btn-group__right': props.right
  })

  return (
    <div className={className}>
      { !!props.title && <div className="btn-group__title">{props.title}</div> }
      <div className="btn-group__inner">
        { props.children }
      </div>
    </div>
  )
}

ButtonGroup.propTypes = {
  title: PropTypes.string,
  right: PropTypes.bool,
  children: PropTypes.arrayOf(PropTypes.element)
}

ButtonGroup.defaultProps = {
  right: false
}

export default ButtonGroup
