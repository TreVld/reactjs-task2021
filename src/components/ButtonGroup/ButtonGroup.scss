.btn-group {
  display: flex;
  align-items: center;
}

.btn-group__inner {
  display: flex;
  align-items: center;
  
  .btn {
    border-radius: 0;
    
    &:first-of-type {
      border-bottom-left-radius: var(--border-radius);
      border-top-left-radius: var(--border-radius);
    }
    
    &:last-of-type {
      border-bottom-right-radius: var(--border-radius);
      border-top-right-radius: var(--border-radius);
    }
  }
}

.btn-group__title {
  opacity: 0.8;
  font-weight: 500;
  text-transform: uppercase;
  margin-right: 1.5rem;
}
