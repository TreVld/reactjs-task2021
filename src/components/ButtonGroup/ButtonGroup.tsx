import React, { ReactNode } from 'react'
import classNames from 'classnames'
import './ButtonGroup.scss'

interface IProps {
  children?: ReactNode
  title?: string
  className?: string
  [index: string]: any
}

function ButtonGroup ({
  children, title, className, ...attrs
}: IProps) {
  const classes = classNames(
    'btn-group',
    className
  )

  return (
    <div className={classes} {...attrs}>
      { !!title && <div className="btn-group__title">{title}</div> }
      <div className="btn-group__inner">{ children }</div>
    </div>
  )
}

ButtonGroup.defaultProps = {
  title: ''
}

export default ButtonGroup
