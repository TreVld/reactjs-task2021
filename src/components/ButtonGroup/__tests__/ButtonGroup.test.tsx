import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'
import ButtonGroup from '../ButtonGroup'

let container: HTMLDivElement | null = null

beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('render ButtonGroup', () => {
  it('should render the content', () => {
    act(() => {
      render(
        <ButtonGroup />,
        container
      )
    })

    expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
