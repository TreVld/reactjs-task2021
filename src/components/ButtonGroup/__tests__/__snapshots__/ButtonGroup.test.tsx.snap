// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ButtonGroup should render the content 1`] = `
"<div class=\\"btn-group\\">
  <div class=\\"btn-group__inner\\"></div>
</div>"
`;
